### ISSUE TITLE
NVDA table navigation does not work in real HTML table when css display:table applied to parent

### ISSUE THREAD

**Issue by manish10** [nvaccess/nvda#11026]
### Steps to reproduce:
In the HTML at the bottom of this issue, navigate to the table on the page and try to use the table navigation keys ctrl+alt+arrow keys. 
### Actual behavior:
NVDA announces not in a table cell.
### Expected behavior:
NVDA should move to the next or previous cell.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
The behavior was the same in older versions of NVDA also.
#### If addons are disabled, is your problem still occuring?
yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
no
``` html
<!DOCTYPE html>
<html xmlns:fo="http://www.w3.org/1999/XSL/Format" lang="en">
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
		<title id="pageTitle">table problems</title>
		<style type="text/css" title="design01">
			.body-container {
				display: table;
			}
		</style>
		</head>
	<body>
	<div class="body-container">

		<h1>test real table inside body of display:css</h1>
		<table>
			<tr>
				<th>C1</th>
				<th>C2</th>
			</tr>
			<tr>
				<td>v1</td>
				<td>v2</td>
			</tr>
			<tr>
				<td>v3</td>
				<td>v4</td>
			</tr>
			</table>
	</div>
	</body>
	</html>
```

**Comment by Adriani90** [comment]
I cannot reproduce this in NVDA Version alpha-20015,56a62097 and Crome or Edge cromium 81. @manish10 could you please test with these verisons?
The last Alpha version of NVDA can be downloaded here:
https://www.nvaccess.org/files/nvda/snapshots/

**Comment by Adriani90** [comment]
For now I am closing this as works for me. Please feel free to comment on this if you are still having this problem and we can reopen it. Thanks.

**Comment by manish10** [comment]
I can reproduce this on chrome and edge with the april 15th alpha build and the latest chrome 81  and edge with the html in the bug.

**Comment by manish10** [comment]
kindly reopen the bug. It is causing a lot of problems with one of our internal apps.

**Comment by Adriani90** [comment]
@manish10 could you please test with NVDA alpha from April 18th?
I am reopening this for now, but I am prety sure this is not reproducible with this Alpha on any of the browsers.
I have created a new html file with your code snippet as attached.
I can open this file in any browser and the table navigation with ctrl+alt+arrow keys works as expected.
Could you please give more details? Do you hear any error in NVDA Alpha while trying to navigate trhough the table?
[Table issue.zip](https://github.com/nvaccess/nvda/files/4499870/Table.issue.zip)

**Comment by Adriani90** [comment]
cc: @feerrenrut

**Comment by manish10** [comment]
The html in the body of the issue got messed up and does not apply the style to the div on the table. apologies for that. I am trying to attach the zip file with the html that will repro the issue. 
[test-table.zip](https://github.com/nvaccess/nvda/files/4500026/test-table.zip)

**Comment by Adriani90** [comment]
Thanks very much, with this file I can reproduce the issue in Crome and Edge.
When I enable reporting of layout tables in NVDA's browse mode settings, following error is written to the log:

```
ERROR - scriptHandler.executeScript (21:47:55.554) - MainThread (11008):
error executing script: <bound method DocumentWithTableNavigation.script_previousRow of <NVDAObjects.IAccessible.chromium.ChromeVBuf object at 0x00C363F0>> with gesture 'Alt+Strg+Pfeiltaste nach oben'
Traceback (most recent call last):
  File "documentBase.pyc", line 157, in _tableMovementScriptHelper
  File "virtualBuffers\gecko_ia2.pyc", line 441, in _getNearestTableCell
  File "documentBase.pyc", line 132, in _getNearestTableCell
LookupError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "virtualBuffers\gecko_ia2.pyc", line 429, in _getTableCellAt
  File "comtypes\__init__.pyc", line 857, in __call__
  File "comtypesMonkeyPatches.pyc", line 26, in __call__
_ctypes.COMError: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "scriptHandler.pyc", line 205, in executeScript
  File "documentBase.pyc", line 175, in script_previousRow
  File "documentBase.pyc", line 163, in _tableMovementScriptHelper
  File "virtualBuffers\gecko_ia2.pyc", line 437, in _getTableCellAt
LookupError
```


I guess the CSS style make it look like a table, but Chrome and Edge might render this as text blocks instead of table cells.
However, maybe @aleventhal can also help here. We saw this issue in different cases and had to close them due to need of external fix from Chromium. Is there maybe already a bug filled with Chromium on this?

In Firefox this works as expected.

**Comment by manish10** [comment]
Both narrator and jaws are able to recognize and work correctly with this table in edge and chrome.
Also, it is a valid semantic table and not made to look as such with css. The css is applied to the parent tag of the table element.

**Comment by bramd** [comment]
I have seen various variants of this bug in the wild as well. It seems Chrome exposes something with `display: table` as a layout table, but since the children of this element have no table semantics at all, this messes up the structure.

I consider it a browser bug that table layouts from CSS get into the accessibility layer and think Firefox handles this correctly these days but had similar issues in the past.

As a web developer, you might consider putting a `role="presentation"` on your `div` with `display:table` to work around this Chromium behavior.

**Comment by manish10** [comment]
Applying role="presentation" does fix the issue.

**Comment by Adriani90** [comment]
Hmm it would be interesting to know why Jaws and Narator can handle the table correctly even without role="presentation".
cc: @leonardder do you have any ideas?

**Comment by Adriani90** [comment]
It seems this is fixed as of NVDA alpha-28126,03c31e5b (2023.2.0.28126)
I tested in Chrome, Edge and Firefox 112 and the table works as expected regardless of layout tables turned on or off in NVDA browse mode settings.

**Comment by Adriani90** [comment]
Closing as works for me.
